core/src/main/java/org/apache/brooklyn/core/sensor/ReleaseableLatch.java

```diff
@@ -75,7 +89,11 @@ public void acquire(Entity caller) {
 
             @Override
             public void release(Entity caller) {
-                sem.release();
+                try {
+                    sem.release();
+                } finally {
+                    ownerEntities.remove(caller);
```

Comment:
Agree, this was my thinking as well. Can always update it when we need to address a more complex use-case.